### PR TITLE
[IOPID-722] Storing prev level on L3 elevation and tax code md5 in storage

### PR DESCRIPTION
**Pull request by claudiodall**
## Short description

The saving of user-related information has been added when attempting to upgrade to L3 
[See the details on confluence.](https://pagopa.atlassian.net/wiki/spaces/IAEI/pages/791774211/IO-ESCO+-+Come+scalare+il+livello+di+accesso+da+L2+a+L3?src=mail&src.mail.timestamp=1694777186040&src.mail.notification=com.atlassian.confluence.plugins.confluence-content-notifications-plugin%3Apage-created-notification&src.mail.recipient=712020%3Adfce7e39-8963-4f34-8190-281ad91fad4b&atlOrigin=eyJpIjoiNjlkYzNlNDgzMGU2NGE1YmFkZGJjNmNkZTRkNzMyNWIiLCJwIjoiYyJ9&src.mail.action=view)

## How to test
On mockoon set /session-state to:

```json
{
  "access_enabled": false,
  "session_info": {
    "active": false,
    "expiration_date":"2025-10-13"
  }
}
```

Run yarn install

Run the project and attempt L2 access then click on "non hai il codice, scopri cosa fare" in order to elevate to L3 using hub spid login. 

Verify that when you return to the portal, the "privilege" object in local storage contains the previous SPID login value and identity (fiscal code MD5 hashed).

Note: The logic for using and deleting the storage is not currently present, it will be implemented in a dedicated branch/task.

### CODE REVIEW

**Comment by shadowsheep1** [comment]
@ebobferraris we need you ;)

File: src/app/[locale]/_component/selectIdp/idpList.tsx [hunk 1]
```diff
@@ -21,7 +25,21 @@ type SpidLevelL3 = {
...
 
 export function IdpList({ spidLevel }: IdpList) {
+  const { userLogged } = useLogin();
+  const token = storageTokenOps.read();
+  const taxCode = token ? userFromJwtToken(storageTokenOps.read()).taxCode : undefined;
+
+  const savePrivilegesData = () => {
+    if (taxCode) {
+      storagePrivilegeOps.write({
+        previous: userLogged?.spidLevel || undefined,
```
Comment (shadowsheep1) [inline src/app/[locale]/_component/selectIdp/idpList.tsx:35]: Is this (`previous`) intentionally named in a cryptic way, or we could give it a more meaningful name (e.g. `previousSecurityLevel`, `previousAccessLevel` or something like that)?

File: src/app/[locale]/_model/Privilege.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { SpidValueInJWT } from './JWTUser';
+
+export type Privilege = {
+  previous: SpidValueInJWT | undefined;
```
Comment (shadowsheep1) [inline src/app/[locale]/_model/Privilege.ts:4]: as above

File: src/app/[locale]/_component/selectIdp/idpList.tsx [hunk 3]
```diff
@@ -21,7 +25,21 @@ type SpidLevelL3 = {
...
+        identity: MD5(taxCode).toString(),
+      });
+    }
+  };
+
   const getSPID = (IDP: IdentityProvider) => {
+    savePrivilegesData();
```
Comment (shadowsheep1) [inline src/app/[locale]/_component/selectIdp/idpList.tsx:42]: I see when we save this date, but I cannot see where we remove this information after a successful login. Will this be the subject of another PR?

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]

File: src/app/[locale]/_component/selectIdp/idpList.tsx (hunk 3 above)
Comment (shadowsheep1) [reply src/app/[locale]/_component/selectIdp/idpList.tsx:42]: I've just seen that this is handled here: https://github.com/pagopa/io-web-profile/pull/47 🙂

File: src/app/[locale]/_component/selectIdp/idpList.tsx (hunk 1 above)
Comment (claudiodall) [reply src/app/[locale]/_component/selectIdp/idpList.tsx]: corrected

File: src/app/[locale]/_model/Privilege.ts (hunk 2 above)
Comment (claudiodall) [reply src/app/[locale]/_model/Privilege.ts]: corrected

**Review by shadowsheep1 — APPROVED** [review]
LGTM

**Pull request merged**